k8s-application/main.tf

```diff
@@ -0,0 +1,29 @@
+resource "mdxc_application_identity" "main" {
+  name = local.params.md_metadata.name_prefix
+}
+
+resource "mdxc_application_permission" "main" {
+  for_each                = local.merged_policies
+  application_identity_id = (mdxc_application_identity.main).gcp_application_identity.service_account_email
```

Comment:
The nested outputs should be mostly for informational purposes if necessary. Ideally we want all of the applications to be able to reference the `mdxc_application_identity.id` and not have to do conditional switching for each cloud.

This means that the `id` thats output in the identity should be whatever the permission attachment needs. (AWS: IAM Role name, GCP: service account email, Azure: service principle ID)

Reply:
ahh okay nice. I'll change the gcp ID to the service account email and that will simplify some of this.